.travis.yml

```diff
@@ -2,12 +2,27 @@ language: c
 sudo: required
 env:
   - TARGET=local
-  - TARGET=docker
+  - TARGET=docker OS=ubuntu-16.04
+  - TARGET=docker OS=fedora-25
 services:
   - docker
 before_install:
-  - if [ ${TARGET}="local" ];  then sudo apt-get update && sudo apt-get install libsqlite3-dev; fi
-  - if [ ${TARGET}="docker" ]; then sudo docker build -t "cangjians/libcangjie:latest" . ; fi
+  - | # before_install: local
+    if [ "${TARGET}" = "local"  ]; then
+      sudo apt-get update && sudo apt-get install libsqlite3-dev
+    fi
+  - | # before_install: docker
+    if [ "${TARGET}" = "docker" ]; then
+      sudo docker pull "cangjians/build-essential:${OS}"
+      sudo docker tag "cangjians/build-essential:${OS}" "cangjians/build-essential:latest"
+      sudo docker build -t "cangjians/libcangjie" .
```

Comment:
Or, if you mean the `cangjians/libcangjie` image, that would not be possible. Reasons:

1. We have to build the image with the source-code-to-test.

2. Also there is a problem of multi-OS. Since I couldn't find a way to override the base image on build (from what I read, it is not possible), I have 3 different `build-essential` images for 3 different OS now. The lines you specified were to manually tag the OS image as `build-essential:latest` so the test may build an image on the OS.